plugin/pkg/global/extensionvalidation/admission.go

```diff
@@ -217,7 +218,7 @@ func (e *ExtensionValidator) Validate(ctx context.Context, a admission.Attribute
 		}
 
 		if !apiequality.Semantic.DeepEqual(shoot.Spec, oldShoot.Spec) {
-			validationError = e.validateShoot(kindToTypesMap, shoot.Spec)
+			validationError = e.validateShoot(kindToTypesMap, computeSupportedForWorkerlessExtensionKindTypes(controllerRegistrationList), shoot.Spec, shoot.IsWorkerless())
```

Comment:
Hm, but then we need to deal with the error message as well. 
Extension is not registered doesn't convey the right error.
Will check again, how it could be improved.